pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -1072,4 +1075,34 @@ private SuccessResponse updateZKTimeIntervalInternal(String tableNameWithType) {
       }
       return new SuccessResponse("Successfully updated time interval zk metadata for table: " + tableNameWithType);
   }
+
+  @GET
+  @Path("/segments/{tableName}/externalViewMismatch")
+  @ApiOperation(value = "Table segment mismatch after rebalance", notes = "Get segments without the same state in EV "
+      + "and IS ")
+  public String getExternalViewSegementMismatch(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "OFFLINE|REALTIME", required = true) @QueryParam("type") String tableTypeStr) {
+    try {
+      TableType tableType = Constants.validateTableType(tableTypeStr);
+
+      if (tableType == null) {
+        throw new ControllerApplicationException(LOGGER, "Table type must not be null", Status.BAD_REQUEST);
+      }
+      String tableNameWithType = TableNameBuilder.forType(TableType.valueOf(tableTypeStr)).tableNameWithType(tableName);
+
+      if (!_pinotHelixResourceManager.hasTable(tableNameWithType)) {
+        throw new TableNotFoundException(String.format("Table=%s not found", tableName));
+      }
+
+      Map<String, MapDifference.ValueDifference<Map<String, String>>> view =
+          _pinotHelixResourceManager.getExternalViewSegementMismatch(tableNameWithType);
+      ObjectNode data = JsonUtils.newObjectNode();
+      data.put("segments", view.toString());
```

Comment:
I feel we might want the response to be the following format (`segmentName` -> `EV/IS` -> different `instanceState`): 
```
{
  "segment0": {
    "IdealState": {
      "server0": "ONLINE",
      "server1": "ONLINE"
    },
    "externalView": {
      "server1": "ERROR",
      "server2": "ONLINE"
    }
  },
  "segment1": {
    "IdealState": {
      "server1": "ONLINE"
    },
    "externalView": {
      "server1": "ERROR"
    }
  }
}
```